sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Exceptions.java

```diff
@@ -54,4 +54,8 @@ public static boolean isPartitionKeyMismatchException(CosmosException cosmosExce
         return Exceptions.isStatusCode(cosmosException, HttpConstants.StatusCodes.NOTFOUND) &&
             Exceptions.isSubStatusCode(cosmosException, HttpConstants.SubStatusCodes.PARTITION_KEY_MISMATCH);
     }
+
+    public static boolean isSDKGeneratedSubStatus(CosmosException cosmosException) {
```

Comment:
I think the implementation is wrong - or at least the behavior I woudl expct when reading the function name is different. Substatuscode 20000-20999 are clinet-side generated errors. 21000-21999 are sever generated (we just add the mapping on the client). I would expect that this method returns false for any substatuscode in the range of 21000-21999 - if the implementation is intentional please rename the function.

Reply:
you are right Fabian, I moved away from using the status code range here as I think it is better to decide case by case.

it is changed into:
- Added a flag within ServiceUnavailableException to distinguish where this is server generated 503 or SDK generated ones
- SDK will only retry if it is server generated 503 or server410 generated 503